logger/noop.go

```diff
@@ -2,8 +2,21 @@ package logger
 
 import (
 	"context"
+	"fmt"
+	"sync"
 )
 
+var once sync.Once
+
 type noopLogger struct{}
 
-func (n noopLogger) Log(context.Context, Entry) {}
+func (n noopLogger) Log(context.Context, Entry) {
+	once.Do(func() {
+		fmt.Println("WARN	[github.com/jacekolszak/yala/logger]	Global YALA logger not set. Please run logger.SetAdapter(printer.StdoutAdapter()) to log messages to stdout.") // nolint
+	})
```

Comment:
problem: warning message will be logger ror local logger too, which is not ideal

Reply:
I created a new dedicated type of noop logger which additional log warning once.